### ISSUE TITLE
AlphaChess version

### ISSUE THREAD

**Issue by Zeta36** [Rochester-NRT/RocAlphaGo#182]
I'm thinking about forking and developing a chess version of this AlphaGo implementation. I know how I should change the main stream of the project, but I would like to know the real state of this work first.

I mean, I can see you already have ready the SL policy and the RL policy (I'm not pretty sure if the RL policy training is working fine or not). I can see anyway that the RL value network training is not yet implemented and I don't know then how do you do the MCTS tree search.

Should I wait for this project to be more advanced before trying to transform it into a chess version? Any suggestion?

Thank you!!

**Comment by Zeta36** [comment]
Hello, @wrongu. I already made some kind of SL policy for chess in this project of mine: https://github.com/Zeta36/Policy-chess

I resolved the problem of themovements by using a fixed labeled vector of legal movements using the "san" Alegraical notation: https://en.wikipedia.org/wiki/Algebraic_notation_(chess)

I extracted the "san" movements from between some millions of games, and it turned out that the amount of "san" movements used are from about 4500 different ones (e4, e5, Bb5, and so on). So I just had to compare between the real player movement and the Softmax predicted movement by the network.

My code (in Tensorflow) is ugly and I made it fast (and furious :P), moreover, the feature planes are not well defined; but even so I could train a SL Policy able to make (without no tree search at all) good openings and something interesting in the middle game but nothing spectacular. 

That's why I thought about using your base code to improve my results with the RL policy training (I was not able to figure out how to implement that part in Tensorflow by my own), but I see you have not advance too much yet in that direction nor in the MCTS implementation, so I guess I'll have to wait for you to finish this part.

I'll be waiting for your improvements ;). And if somebody wants to play with my ugly Chess (TensorFlow) SL policy network will be of course welcome.

Regards.

**Comment by Zeta36** [comment]
@wrongu, the reinforcement policy training step (reinforcement_policy_trainer.py) is working right now? I mean, do you have already the SL and RL policy working and operative? 
What's the current state of the project? If I try converting the project into chess would I be able to use RL to improve the SL model?

**Comment by wrongu** [comment]
@Zeta36 RL appears to be functional now. The tests were made much more exhaustive since you first saw it, which exposed a few bugs that have since been fixed. SL has been functional for a while, but it's being cleaned up now (see #193). We will likely do something similar to RL soon.

I've done some RL training - enough to verify at least that the network isn't _unlearning_ everything (which it used to do), but not enough yet to see real improvements in play strength.
